### ISSUE TITLE
Gmail API Code sample is wrong?

### ISSUE THREAD

**Issue by xingped** [google/google-api-nodejs-client#648]
I was using the Gmail API code sample here: https://developers.google.com/gmail/api/quickstart/nodejs

And it seems the node client expects the client_secret.json file in a format that is different from what is given when registering for the API here: https://console.developers.google.com/start/api?id=gmail

When the credentials file is loaded, the code sample looks for `credentials.installed.client_secret` among other properties, but the structure of the client_secret.json file is as such:

    {
      "type": "service_account",
      "project_id": "...",
      "private_key_id": "...",
      "private_key": "...",
      "client_email": "...",
      "client_id": "...",
      "auth_uri": "https://accounts.google.com/o/oauth2/auth",
      "token_uri": "https://accounts.google.com/o/oauth2/token",
      "auth_provider_x509_cert_url": "...",
      "client_x509_cert_url": "..."
    }

So yeah, bug report is that the code sample is broken.

Since I'm already here though. Referencing the code sample, which properties should be used for the clientSecret, clientId, and redirectUrl if the code sample is wrong?

**Comment by jmdobry** [comment]
It looks like you created a Service Account key when the sample calls for an OAuth2 Client ID. Service Accounts are typically used for server-to-server API calls. GMail needs an OAuth2 Client ID, as it typically requires that a user authorize your code to act on the user's behalf.

Here's a direct link to creating an OAuth2 Client ID: https://console.cloud.google.com/apis/credentials/oauthclient

**Comment by alan-mroczek** [comment]
In my case similar to that @xingped, I had to change installed to web, but there is possibility that this was my fault, however you can check that,

**Comment by jmdobry** [comment]
Switch from a Service Account to an OAuth2 Client ID, and you should be good to go.

**Comment by j-gironda** [comment]
Hi @xingped , did you fix it? I'm stuck with the same problem. 
I generated the json following the google documentation instruction. Mi goal is to upload/update apks from a java application that will be executed from a jenkins machine.

I got a json with the same structure as yours, but when using the google examples it always fails because the json doesn't have the clien_secret field (nor the installed object wrapping it).

Thanks in advance.
